File: player/ultraPlayer/RemoveAddonRequest.cpp
#include "RemoveAddonRequest.h"

RemoveAddonRequest::RemoveAddonRequest(void):ARequest(MessageType::RemoveAddon)
{
}

RemoveAddonRequest::~RemoveAddonRequest(void)
{
}

void RemoveAddonRequest::FromString( string data )
{
	try
	{
		int index = CommunicationProtocol::Instance()->HeaderLength();
		_addonId = NumericToStringConverter::Convert<UINT>(data.substr(index, CommunicationProtocol::Instance()->NumericParameterLength()));

	}
	catch (exception& e)
	{
		throw ParserException(e.what(), "RemoveAddonRequest::FromString");
	}
}

UINT RemoveAddonRequest::AddonId()
{
	return _addonId;
}

File: controller/Model/Communication/Request/RemoveAddonRequest.cs
using System;
using System.Collections.Generic;
using System.Text;
using UltraPlayerController.Model.Enumerations;
using UltraPlayerController.Communication;
using UltraPlayerController.Resources.Communication;
using UltraPlayerController.Model.Utilities;
using Utilities.Serialize;

namespace UltraPlayerController.Model.Communication.Request
{
    class RemoveAddonRequest : ARequest
    {
        #region Fields
        private uint _addonId;
        #endregion

        #region Constructors
        public RemoveAddonRequest(): base(MessageType.RemoveAddon)
        {

        }
        #endregion

        #region ARequest members
        public override string ParseToString()
        {
            string request = string.Empty;

            try
            {
                request += base.ParseToString();

                request += ValueSerializer.DoInt(Protocol.Instance.NumericParameterLength, (int)_addonId);
            }
            catch (System.Exception)
            {
            	
            }
            return request;
        }
        #endregion

#region Properties
	public uint AddonId
    {
        set
        {
            _addonId = value;
        }
    }
#endregion
    }
}
